[docs] Update Push notifications and expo-notifications docs (#20270)
expo/expo: docs/pages/push-notifications/receiving-notifications.mdx

---
title: Receive notifications
---

import { BoxLink } from '~/ui/components/BoxLink';

> Handling notifications with Expo is straightforward across all platforms. You can use event listeners from [`expo-notifications`](/versions/latest/sdk/notifications) library to handle how your app responds when receiving a notification and how to set the behavior of the push notifications when your app is in the foreground.

Reviewer: ```suggestion
The [`expo-notifications`](/versions/latest/sdk/notifications) library contains event listeners that handle how your app responds when receiving a notification.
```

## Notification event listeners

The `addNotificationReceivedListener` and `addNotificationResponseReceivedListener` event listeners receive an object when a notification is received or interacted with.

> These two different listeners allow you to address cases where a notification comes in while your app is open and foregrounded, **and** cases where a notification comes in while your app is backgrounded or closed, and the user taps on the notification.

Reviewer: ```suggestion
These listeners allow you to add behavior when notifications are received while your app is open and foregrounded and when your app is backgrounded or closed and the user taps on the notification.
```

  responseListener.current = Notifications.addNotificationResponseReceivedListener(response => {
    console.log(response);
  });
  /* @end */

  return () => {
    Notifications.removeNotificationSubscription(notificationListener.current);
    Notifications.removeNotificationSubscription(responseListener.current);
  };
}, []);
```

For more information on these objects, see [`Notification`](/versions/latest/sdk/notifications/#notification) documentation.

## Foreground notification behavior

To handle the behavior when notifications are received when your app is **foregrounded**, use [`Notifications.setNotificationHandler`](/versions/latest/sdk/notifications/#handling-incoming-notifications-when-the-app-is) with the `handleNotification()` callback to set the following options:

- `shouldShowAlert`
- `shouldPlaySound`
- `shouldSetBadge`

```jsx
Notifications.setNotificationHandler({
  handleNotification: async () => ({
    shouldShowAlert: true,
    shouldPlaySound: false,
    shouldSetBadge: false,
  }),
});
```

## Closed notification behavior

> On Android, users can set certain OS-level settings (**usually** revolving around performance and battery optimization), that can prevent notifications from being delivered when the app is closed. For example, one such setting is the **Deep Clear** option on OnePlus devices using Android 9 and lower versions.

Reviewer: ```suggestion
On Android, users can set certain OS-level settings, usually revolving around performance and battery optimization, that can prevent notifications from being delivered when the app is closed. For example, one such setting is the **Deep Clear** option on OnePlus devices using Android 9 and lower versions.
```
